core/src/components/picker-internal/picker-internal.tsx

```diff
@@ -0,0 +1,504 @@
+  private enterInputMode = (columnEl?: HTMLIonPickerColumnInternalElement, focusInput = true) => {
+    const { inputEl, el } = this;
+    if (!inputEl) { return; }
+
+    /**
+     * Only active input mode if there is at
+     * least one column that accepts numeric input.
+     */
+    const hasInputColumn = el.querySelector('ion-picker-column-internal.picker-column-numeric-input');
+    if (!hasInputColumn) { return; }
+
+    /**
+     * If columnEl is undefined then
+     * it is assumed that all numeric pickers
+     * are eligible for text entry.
+     * (i.e. hour and minute columns)
+     */
+    this.inputMode = true;
+    this.inputModeColumn = columnEl;
+
+    /**
+     * Users with a keyboard and mouse can
+     * activate input mode where the input is
+     * focused as well as when it is not focused,
+     * so we need to make sure we clean up any
+     * old listeners.
+     */
+    if (focusInput) {
+      if (this.destroyKeypressListener) {
+        this.destroyKeypressListener();
+        this.destroyKeypressListener = undefined;
+      }
+
+      inputEl.focus();
+    } else {
+      el.addEventListener('keypress', this.onKeyPress);
+      this.destroyKeypressListener = () => {
+        el.removeEventListener('keypress', this.onKeyPress);
+      }
+    }
+
+    this.emitInputModeChange();
+  }
+
+  private exitInputMode = () => {
+    const { inputEl, inputMode } = this;
+    if (!inputMode || !inputEl) { return; }
+
+    this.inputMode = false;
+    this.inputModeColumn = undefined;
+    inputEl.blur();
+    inputEl.value = '';
+
+    if (this.destroyKeypressListener) {
+      this.destroyKeypressListener();
+      this.destroyKeypressListener = undefined;
+    }
+
+    this.emitInputModeChange();
+  }
+
+  private onKeyPress = (ev: KeyboardEvent) => {
+    const { inputEl } = this;
+    if (!inputEl) { return; }
+
+    const parsedValue = parseInt(ev.key, 10);
+
+    /**
+     * Only numbers should be allowed
+     */
+    if (!Number.isNaN(parsedValue)) {
+      inputEl.value += ev.key;
+
+      this.onInputChange()
+    }
+  }
+
+  private selectSingleColumn = () => {
+    const { inputEl, inputModeColumn, singleColumnSearchTimeout } = this;
+    if (!inputEl || !inputModeColumn) { return; }
+
+    const values = inputModeColumn.items;
+    let valueToSelect;
+
+    /**
+     * If users pause for a bit, the search
+     * value should be reset similar to how a
+     * <select> behaves. So typing "34", waiting,
+     * then typing "5" should select "05".
+     */
+    if (singleColumnSearchTimeout) {
+      clearTimeout(singleColumnSearchTimeout);
+    }
+
+    this.singleColumnSearchTimeout = setTimeout(() => {
+      inputEl.value = '';
+      this.singleColumnSearchTimeout = undefined;
+    }, 1000);
+
+    /**
+     * For values that are longer than 2 digits long
+     * we should shift the value over 1 character
+     * to the left. So typing "456" would result in "56".
+     */
+    if (inputEl.value.length >= 3) {
```

Comment:
I'll put a TODO for now so that if we want to add this support we know to update this. Would rather not add the functionality right now since it would mean looking at every value for every column on load.